src/test/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/MeanQualityByCycleSparkIntegrationTest.java

```diff
@@ -0,0 +1,35 @@
+package org.broadinstitute.hellbender.tools.spark.pipelines.metrics;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.tools.IntegrationTestSpec;
+import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public final class MeanQualityByCycleSparkIntegrationTest extends CommandLineProgramTest {
+    private static final File TEST_DATA_DIR = new File(getTestDataDir(), "picard/analysis/MeanQualityByCycle");
+
+    @Override
+    public String getTestedClassName() {
+        return MeanQualityByCycleSpark.class.getSimpleName();
+    }
+
+    @Test
+    public void test1() throws Exception {
+        //Note we compare to non-spark outputs
+        final File unsortedBam = new File(TEST_DATA_DIR, "first5000a.bam");
+        final File expectedFile = new File(TEST_DATA_DIR, "meanqualbycycle.txt");
+        final File outfile = BaseTest.createTempFile("testMeanQualityByCycle", ".metrics");
+        ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--" + StandardArgumentDefinitions.INPUT_LONG_NAME);
+        args.add(unsortedBam.getCanonicalPath());
+        args.add("--" + StandardArgumentDefinitions.OUTPUT_LONG_NAME);
+        args.add(outfile.getCanonicalPath());
+        this.runCommandLine(args.getArgsArray());
+
+        IntegrationTestSpec.assertEqualTextFiles(outfile, expectedFile, "#");
```

Comment:
There should be test case that includes `--ALIGNED_READS_ONLY` and one with `--PF_READS_ONLY`